patches/react-native-share-menu+6.0.0.patch

```diff
@@ -1,13 +1,94 @@
+diff --git a/node_modules/react-native-share-menu/android/build/intermediates/javac/debug/classes/com/meedan/ShareMenuModule.class b/node_modules/react-native-share-menu/android/build/intermediates/javac/debug/classes/com/meedan/ShareMenuModule.class
+new file mode 100644
+index 0000000..38173b7
+Binary files /dev/null and b/node_modules/react-native-share-menu/android/build/intermediates/javac/debug/classes/com/meedan/ShareMenuModule.class differ
+diff --git a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+index 09abd7b..5121398 100644
+--- a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
++++ b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+@@ -112,6 +112,14 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
+     clearSharedText();
+   }
+ 
++  @ReactMethod
++  public void addListener(String eventName) {
++  }
++
++  @ReactMethod
++  public void removeListeners(Integer count) {
++  }
++
+   private void dispatchEvent(ReadableMap shared) {
+     if (mReactContext == null || !mReactContext.hasActiveCatalystInstance()) {
+       return;
+diff --git a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+index e290cce..44e5dfe 100644
+--- a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
++++ b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+@@ -34,6 +34,8 @@ public class ShareMenuReactView: NSObject {
+             return
+         }
+ 
++        ShareMenuReactView.detachViewDelegate()
++
+         if error != nil {
+             let exception = NSError(
+                 domain: Bundle.main.bundleIdentifier!,
 diff --git a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
-index f42bce6..ee36062 100644
+index f42bce6..ab00813 100644
 --- a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 +++ b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
 @@ -13,7 +13,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
    func sourceURL(for bridge: RCTBridge!) -> URL! {
  #if DEBUG
      return RCTBundleURLProvider.sharedSettings()?
 -      .jsBundleURL(forBundleRoot: "index.share", fallbackResource: nil)
-+      .jsBundleURL(forBundleRoot: "index.share")
++      .jsBundleURL(forBundleRoot: "index")
```

Comment:
Is there a reason anyone might ever need to use index.share.js?

Reply:
Kind of comes down to code organization preferences. It makes it easier/more declarative to specifically setup the share extension. In most apps that would be preferred. In our case since we have a ludicrously tangled mess it makes more sense to just reuse index.js.